src/app.py

```diff
@@ -50,6 +51,41 @@ def convert_str_to_bool(str: str) -> bool:
     except AttributeError as err:
         logger.error(err)
 
+def replace_str_in_file(file_path: str, find: str, replace: str):
+    """
+    Replace string in a file.
+
+    :param file_path: target file path
+    :param find: string to find
+    :param replace: string to replace
+    """
+    # Read in the file
+    with open(file_path, 'r') as file :
+        filedata = file.read()
+
+    # Replace the target string
+    filedata = filedata.replace(find, replace)
+
+    # Write the file out again
+    with open(file_path, 'w') as file:
+        file.write(filedata)
+
+def get_selenium_jar_url():
+    """
+    Get the latest selenium server jar url 
+    """
+    # Command to find jar url
+    command = 'curl -sk https://github.com/SeleniumHQ/selenium/releases/ | grep ".jar" | head -n 1 | grep -Eo "/[a-zA-Z0-9./?=_%:-]*"'
```

Comment:
we always need to install specific version instead of installing the latest.

Reply:
Downloading selenium jar file not work if jar file not in visible part of assets. So as wrote @budtmo need set version
![image](https://user-images.githubusercontent.com/429667/221805713-e56e036f-f993-4191-af5c-e35d7a9e8aa0.png)

Reply:
Thank you @budtmo @SHELA, yes, you are both right, by the time I created the code to download the jar file those assets had included the `.jar` extension, but now it was changed to `.zip` extensions. I will be refactoring this downloading process and set to one specific version.

Reply:
> I dont know why we need to install selenium.jar? is it like client to be able to connect to selenium hub? it need to be in the same container or it can be in different container?

Yes, in selenium 4 version things changed a bit. We definitely need the jar file to create the connection between the hub and the node as it is specified here at the [ appium conference - Selenium Grid 4 and Appium together in harmony](https://www.youtube.com/watch?v=3_aP2rsqZD0) 
![Screen Shot 2023-03-02 at 20 07 19](https://user-images.githubusercontent.com/33426940/222640266-9d686228-b0f2-4303-821f-f61bebf2339f.png)

Also, is worth mentioning that I'm using the same approach to create this connection using [this code](https://github.com/SeleniumHQ/docker-selenium/blob/trunk/NodeDocker/start-selenium-grid-docker.sh) as a reference from the official [docker-selenium repository ](https://github.com/SeleniumHQ/docker-selenium). But I'm open to suggestions or improvements on this, please let me know if it makes sense to you, otherwise, we could try to come up with another solution.

Reply:
hm... @diemol could you maybe help me by explaining if it is possible that Selenium agent to be different container / image or is there any plan to do it? e.g. Selenium agent run on different container and we just need to point out where the Appium server is running through url.

if it is not possible:
1. I will just merge this PR because anyway, I am working on big refactoring on this project and will change things in this PR to fit the new code structure. 
2. for Selenium, I will provide extended docker image that has Selenium agent and will support only Selenium 4.x

Reply:
Selenium agent? What do you mean? @budtmo

Reply:
sorry @diemol for late response. hm... do I need to download selenium.jar inside this docker image to connect to docker hub (from another container)? if yes, is the jar file need to be inside the same docker container? btw, is there any documentation that I can have a look how to connect Appium to Selenium 4? :/

Reply:
thank you so much @diemol . it seems the selenium.jar can be located in different machine / docker container because it just read the whole config from .toml that has the Appium url inside it. 

last question: does docker-selenium provide simple docker image to act as custom "node" to read the .toml file? I cannot find it in docker-selenium repo and it seems that I cannot extend the NodeBase docker-image as it has things like xvfb etc that we dont need. If I create this "custom node" inside docker-selenium, where it should go?

Reply:
No, we do not have it in the docker-selenium repository because it is not a common use case. And we just do not want to increase the maintenance. Probably the easiest for you now is to add it to the image?

Reply:
So, @budtmo your major concern is downloading and using the jar file inside the android container, do you think the jar size will affect the performance of the android emulator inside the container? If that is the case we can maybe use another container just to store the jar file, or as @diemol said, add that jar file to the android docker image?

Reply:
@LerryAlexander it is not about performance. I am still working on the big refactoring on this project, so I try to avoid any dependency with other tools to make it easy to maintain. if it is about normal configuration like Selenium 3, I am fine with it, but if it is about downloading selenium.jar, it need to be actively updated as well because e.g. user uses the newest Selenium Hub from docker-selenium and selenium.jar in this project is the previous version which might not compatible anymore with that newest version of Hub. that why I think the "custom node" should go to docker-selenium repository. 

but as @diemol have a hard time to find contributors to help to maintain Selenium repository, I can probably add simple docker image for that selenium.jar only either in this project or I will send PR to docker-selenium or you want to do it @LerryAlexander?

Reply:
mmm ok, I get it @budtmo. 
Ok, yes, I could create a simple docker image to store the selenium.jar file. I could include that change in this PR and from there then we can decide if it is better to add it only to this project or create a PR on the docker-selenium repo. If I have any doubts and need you to re-check I'll reach out to you, Sounds good?

Reply:
Hi @LerryAlexander , 

lets put the whole Selenium integration inside [appium-docker-android](https://github.com/appium/appium-docker-android) project. the scripts can go inside new folder "Selenium-Node". I just created a new docker image repo with name [appium/selenium-node](https://hub.docker.com/repository/docker/appium/selenium-node/general). I think it make more sense. what do you think? if you are agree, I will close this PR and we move to appium-docker-android project.

Reply:
You can also extend the docker selenium base image.

Reply:
lets do it inside docker-selenium then @LerryAlexander , @diemol will review your PR. I will close this PR then.

Reply:
My PR? I got lost 😄

Reply:
hahaha.. :D  sorry, I meant @LerryAlexander might create a PR to that docker-selenium and it will be reviewed by you @diemol

Reply:
Sorry @budtmo I'm lost here. I believe we still need the script proposed in this PR, or maybe I didn’t get you clearly enough. So, when the user is using docker-android image and wants to connect to selenium hub using selenium grid, from the docker-android image I was proposing to use the env variable `CONNECT_TO_GRID_4` to run all the script proposed in the PR; so now that script should be moved to docker-selenium repo? Sorry, I'm not getting the logic here, help me to clarify the next steps.

Reply:
@LerryAlexander if I understand correctly based on the documentation: to be able to connect to Selenium 4.x, you just need to run selenium.jar with .toml file that contains Appium url and because we just need selenium.jar and .toml file, we can create a small Dockerfile in docker-selenium project. it means, we dont need anything to be done here, or do you mean something else? or maybe I miss something?

Reply:
Hello, sorry for the late reply @budtmo. Yes, you are right, but the whole idea of this PR was using the docker-android image to be able to create the selenium grid architecture just like the way you mentioned in your [docker-compose file exampe](https://github.com/budtmo/docker-android/blob/master/docker-compose.yml) but this time using Selenium 4.x, because right now the current script you have `src/app.py` works using the old architecture. So, I know that the approach of downloading the .jar file is a requisite to connect no Selenium 4.x, but besides having the toml file, so, for example, we need to get the `platformVersion` or the `browserName` from the environment variables provide to the docker-android image to inject the `configs` section info as you can see in the toml file below: 

```
[node]
detect-drivers = false

[relay]
# Default Appium/Cloud server endpoint
url = "http://localhost:4723/wd/hub"
status-endpoint = "/status"
# Stereotypes supported by the service. The initial number is "max-sessions", and will allocate 
# that many test slots to that particular configuration
configs = [
  "5", "{\"browserName\": \"chrome\", \"platformName\": \"android\", \"appium:platformVersion\": \"11\"}"
]
```

So, all this connection is happening in android-docker image, it's just that we are updating the image to connect now to Selenium 4.x instead of the old Selenium 3.x architecture, so I don’t see how creating an image in docker-selenium could help us to use this whole approach, do you get my point? 
I mean, I can create the docker image in selenium repo that does the thing about downloading the selenium.jar file and the toml configuration, but how can we relate that using the docker android image which spins up the emulator inside the container? Maybe I'm missing something, can you explain to me how can we use docker-android with selenium 4.x if we move this whole logic to a Dockerfile in docker-selenium project?